src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/DecimalMathFuncs.txt

```diff
@@ -0,0 +1,453 @@
+
+// decimal aware functions with mixed inputs
+
+>> Int( [1000000000000000000000001.1, 2000000000000000000000002.2, Float(100000.12)] )
+Table({Value:1000000000000000000000001},{Value:2000000000000000000000002},{Value:100000})
+
+>> Trunc( [1000000000000000000000001.1, 2000000000000000000000002.2, Float(100000.12)] )
+Table({Value:1000000000000000000000001},{Value:2000000000000000000000002},{Value:100000})
+
+>> Round( [1000000000000000000000001.1, 2000000000000000000000002.2, Float(300000.1234)], 2 )
+Table({Value:1000000000000000000000001.1},{Value:2000000000000000000000002.2},{Value:300000.12})
+
+>> RoundUp( [1000000000000000000000001.1, 2000000000000000000000002.2, Float(300000.1234)], 2 )
+Table({Value:1000000000000000000000001.1},{Value:2000000000000000000000002.2},{Value:300000.13})
+
+>> RoundDown( [1000000000000000000000001.1, 2000000000000000000000002.2, Float(30000.1234)], 2 )
+Table({Value:1000000000000000000000001.1},{Value:2000000000000000000000002.2},{Value:30000.12})
+
+// these rounding tests are returning the correct results, due to the conversion between decimal and float
+
+>> Int( [Float(100000.12), 1000000000000000000000001.1, 2000000000000000000000002.2] )
+Table({Value:100000},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+>> Trunc( [Float(100000.12), 1000000000000000000000001.1, 2000000000000000000000002.2] )
+Table({Value:100000},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+>> Round( [Float(300000.1234), 1000000000000000000000001.1, 2000000000000000000000002.2], 2 )
+Table({Value:300000.12},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+>> RoundUp( [Float(300000.1234), 1000000000000000000000001.1, 2000000000000000000000002.2], 2 )
+Table({Value:300000.13},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+>> RoundDown( [Float(30000.1234), 1000000000000000000000001.1, 2000000000000000000000002.2], 2 )
+Table({Value:30000.12},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+// non-decimal aware numeric functions
+
+>> Sqrt( 1524157899404237160815376 )         // exact square of 1234567899876, should be done in float and match next test
+1234567899876
+
+>> Sqrt( Float(1524157899404237160815376) )  // exact square of 1234567899876
+1234567899876
+
+>> Sqrt( 4.000000000000000004 )              // outside of float range, will round to 4
+2
+
+>> Sqrt( 4.0000000004 )                      // within float range
+2.0000000001
+
+>> Log( 100, 10 )
+2
+
+>> Log( 10000000000000050000000, 10 )         // input will be rounded with float conversion
+22
+
+>> Log( 10000000000000050000000, Float(10) )
+22
+
+>> Log( 10000000000000000000000, 10 )         // above tests should match these answers
+22
+
+>> Log( 10000000000000000000000, Float(10) )  
+22
+
+>> Log( Float("1e100"), 10 )
+100
+
+>> Ln( 2.71828182845904523536028747135266249775724709369995 )
```

Comment:
>Ln( 2.71828182845904523536028747135266249775724709369995 ) [](http://example.com/codeflow?start=3&length=58)

I'd try Ln( 2.71828182845904523536028747135266249775724709369995 ) - 1

Reply:
Comes back with 0.  :)  I'm not trying to identify tiny differences, instead looking for consistency between what decimal and float would return.  And consistency with SQL.  In this case it is all float so it shouldn't matter much.